src/components/WeatherWidget/WeatherWidget.tsx

```diff
@@ -0,0 +1,311 @@
+
+  const { forecast, getForecast } = useWeather()
+  const { updateModalId, updateModalText, modalRef, updateWidget } =
+    useModalContext()
+
+  const {
+    addNewWeatherWidget,
+    editWeatherWidget,
+    isAddingWidget,
+    setIsAddingWidget,
+    setDisableDragAndDrop,
+  } = useMySpaceContext()
+
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  // WeatherWidget settings dropdown state
+  const dropdownEl = useRef<HTMLDivElement>(null)
+  const [isDropdownOpen, setIsDropdownOpen] = useCloseWhenClickedOutside(
+    dropdownEl,
+    false
+  )
+
+  useEffect(() => {
+    if (widget.widget) {
+      getForecast(widget.widget.coords.hourlyForecastUrl)
+    }
+  }, [widget.widget])
+
+  useEffect(() => {
+    if (isAddingWidget || isEditing) {
+      inputRef?.current?.focus()
+    }
+  }, [isAddingWidget, isEditing])
+
+  const currentForecast = forecast.slice(0, 5)
+
+  const inputId = widget.widget
+    ? `weatherWidget_${widget.widget._id}`
+    : 'weatherWidget_temporary'
+
+  /** Remove widget */
+  // Show confirmation modal
+  const handleConfirmRemoveWidget = () => {
+    if (isAddingWidget && !widget.widget) {
+      setIsAddingWidget(false)
+    }
+
+    if (!isAddingWidget && widget.widget) {
+      updateModalId('removeNewsWidgetModal')
+      updateModalText({
+        headingText: 'Are you sure you’d like to delete this widget?',
+        descriptionText:
+          'You can re-add it to your My Space from the Add Widget menu.',
+      })
+
+      const widgetState: WeatherWidgetType = {
+        _id: widget.widget._id,
+        title: widget.widget.title,
+        type: 'Weather',
+        coords: widget.widget.coords,
+      }
+
+      updateWidget(widgetState)
+
+      modalRef?.current?.toggleModal(undefined, true)
+    }
+
+    setDisableDragAndDrop(false)
+  }
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    const data = new FormData(e.currentTarget)
+    const inputVal = `${data.get('weatherWidget_input')}`
+
+    if (isAddingWidget && !widget.widget) {
+      addNewWeatherWidget(inputVal.toString())
+      setIsAddingWidget(false)
+    }
+
+    if (isEditing && widget.widget) {
+      setIsEditing(false)
+      editWeatherWidget({
+        _id: widget.widget._id.toString(),
+        zipcode: inputVal.toString(),
+      })
+    }
+
+    setDisableDragAndDrop(false)
+  }
+
+  const handleCancel = () => {
+    // Edge case: what if someone is adding another widget while editing this widget? Might need to add a
+    // conditional here.
+    setIsAddingWidget(false)
+    setIsEditing(false)
+    setDisableDragAndDrop(false)
+  }
+
+  const handleEdit = () => {
+    setIsEditing(true)
+    setIsDropdownOpen(false)
+    setDisableDragAndDrop(true)
+  }
+
+  const handleKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    const keyCode = e.key
+
+    // Allow only digits (0-9)
+    if (!/^\d$/.test(keyCode)) {
+      e.preventDefault()
+    }
+  }
```

Comment:
I had to add this because, even after adding a pattern to the input and restricting it to 5 digits, you could still type special characters for some reason. So this function intercepts each key press, and if it isn't a single digit, it will not allow the value to be placed in the input field.